Deployment.md refreshing (#801)
btcpayserver/btcpayserver-doc: docs/Deployment.md

> Developer deployments are not recommended for production environments and require the user to have technical knowledge related to the build.

Reviewer: > Developer deployments
It's unclear what developer deployments are?

Author: ```suggestion
Manual deployments and Hardware builds are not recommended for production environments and require the user to have technical knowledge related to the build.
```

This should make it more understandable, and more relating to the titles `Manual deployment` and `Hardware build` shown in the new table.
Thoughts ?

Reviewer: Yes, that's clearer indeed. :)

Author: Then committing that :)

:::

## To chose one that will best suit your needs, consider the following:<br>

> |Choosing a deployment <br>method      |   Difficulty   |    Production/Dev |      All features <br>available     |             Approx. cost<br> per month      |

Reviewer: ```suggestion
|Deployment method <br>method      |   Difficulty   |    Production/Dev |      All features <br>available     |             Approx. cost<br> per month      |
```

> | -------------------------------- |:--------------:|:----------------------:|:---------------:|:-------------:|:-------------:|

Reviewer 2: ```suggestion
| -------------------------------- |:--------------:|:----------------------:|:---------------:|:-------------:|
```

The github preview of this table doesn't look right. Are you sure you need 6 segments here with only 5 columns?

Author: Oh I must have forgotten to delete that. I had a 6th column before making this PR.
